data-serving/data-service/api/openapi.yaml

```diff
@@ -221,6 +258,23 @@ components:
           type: array
           items:
             $ref: "#/components/schemas/Case"
+    NewCase:
+      description: A "#/components/schemas/Case" with additional required fields for newly-created cases
+      allOf:
+        - $ref: "#/components/schemas/Case"
+      properties:
+        caseReference:
+          required:
+            - sourceId
+            - sourceUrl
+        location:
+          required:
+            - country
+      required:
+        - caseReference
+        - location
+        - events
```

Comment:
I can't imagine there's a good way to encode requiring an event with a specific `name` field. Is there a description portion where it'd be easy to clarify this?

Reply:
Yeah fair, I will add it to the `description` for this property in the next PR when I flesh out the `events` prop